projects/Mallard/src/services/logging.ts

```diff
@@ -158,6 +158,18 @@ class Logging {
         }
     }
 
+    // Designed to post logs that have been queued but havent sent
+    async postQueuedLogs(): Promise<void> {
+        try {
+            const queuedLogsString = await this.getQueuedLogs()
+            const queuedLogs = JSON.parse(queuedLogsString)
+            await this.postLog(queuedLogs)
+        } catch {
+            // Assumes there is a problem sending logs and clears them
+            await this.clearLogs()
```

Comment:
We can be a little more defensive, `clearLogs` only be called when network available? In offline mode, it will always fail and we may lose some valuable logs.

Reply:
Yep - good shout. Will modify